src/components/ActionRow.ts

```diff
@@ -0,0 +1,50 @@
+import { APIActionRowComponent, ComponentType } from 'discord-api-types/v9';
+import type { LinkButtonComponent, InteractionButtonComponent, SelectMenuComponent } from '..';
+import type { Component } from './Component';
+import { createComponent } from './Components';
+
+export type ActionRowComponent = LinkButtonComponent | InteractionButtonComponent | SelectMenuComponent;
+
+// TODO: Add valid form component types
+
+/**
+ * Represents an action row component
+ */
+export class ActionRow<T extends ActionRowComponent> implements Component {
+	public readonly components: T[] = [];
+	public readonly type = ComponentType.ActionRow;
+
+	public constructor(data?: APIActionRowComponent) {
+		if (!data) {
+			return;
+		}
+
+		this.components = data.components.map(createComponent) as T[];
```

Comment:
Don't return in constructors, use `this.components = data?.components.map ?? []`